main/src/library/MutList.flix

```diff
@@ -18,145 +18,412 @@
 /// Represents a mutable list.
 ///
 pub enum MutList[a] {
-    case MutList(Array[a], Int32)
+    case MutList(Ref[Array[a]], Ref[Int32])
 }
 
 namespace MutList {
 
-    /*
-     * Creation operations.
-     */
-
     ///
     /// Returns a new empty mutable list with a default capacity.
     ///
-    pub def new(): MutList[a] = newWithCapacity(0)
+    pub def new(): MutList[a] & Impure = MutList(ref [], ref 0)
 
     ///
     /// Returns a new empty mutable list with the given `capacity`.
     ///
-    pub def newWithCapacity(capacity: Int32): MutList[a] = ???
+    pub def newWithCapacity(c: Int32): MutList[a] & Impure = ??? /// TODO: How to allocate array without an element type?
```

Comment:
Yeah, I think this is currently *impossible* to implement. Let us just get rid of `newWithCapacity` for now.